Foundation/NSOrderedSet.swift

```diff
@@ -9,23 +9,26 @@
 
 /****************       Immutable Ordered Set   ****************/
 open class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, ExpressibleByArrayLiteral {
-    internal var _storage: Set<NSObject>
-    internal var _orderedStorage: [NSObject]
+    internal var _storage: NSMutableSet
+    internal var _orderedStorage: NSMutableArray
```

Comment:
this is essentially how the ObjC-based Darwin implementations are handled, so thats "A Nice Thing™"

Reply:
Actually looking more closely -- I realize this is the NSOrderedSet impl -- why are these mutable?

Reply:
Good catch - these no longer need to be vars

Reply:
Although it does allow internal implementation to copy the `_storage` from one set to another, like when implementing `copy`

Reply:
Sorry, I see I misunderstood your question. I could override these in `NSMutableOrderedSet` to be mutable and keep these ones as their non-mutable counter parts, but keeping these as mutable makes it easier to build the set incrementally in initializers

Reply:
Building incrementally isn't the best for performance (sans capacity hints that are obeyed) - since you'll hit the internal vector growth storage of array/set many times.